### PR TITLE
`hypothesis` tests; numba-based fixes for `Chandrasekhar_G`

### PR DESCRIPTION
**Pull request by StanczakDominik**
From #1059. This PR:

* adds tests for `Chandrasekhar_G` using [`hypothesis`](https://hypothesis.readthedocs.io/); **I would like to use this as an example for the tutorial I was supposed to give yesterday and will give at the telecon sometime soon.** So I suspect it will become clearer what it does then. I suspect we'll record that, too. Thus, this is reviewable, but with that caveat.
* turns `Chandrasekhar_G` into a numpy `ufunc` using `numba.vectorize`. This lets me handle the cases of very small and very large arguments cleanly, while allowing application to numpy arrays, without the annoyances of floating point numerical errors.
* uses https://github.com/numba/numba-scipy 's implementation of `erf`. It does so implicitly through numba's clever overload mechanisms. I just import the package and it registers its overload for `scipy.special.erf`

### CODE REVIEW

**Review by namurphy — APPROVED** [review]
Looks good to me!  I have a few mostly inconsequential comments/suggestions, but it'd be perfectly fine to merge as is.  Thanks!

File: plasmapy/formulary/mathematics.py [hunk 1]
```diff
@@ -131,7 +139,7 @@ def Chandrasekhar_G(x: float):
     >>> Chandrasekhar_G(1)
     0.21379664776456
     >>> Chandrasekhar_G(1e-6)
-    3.7608262858090935e-07
+    3.7602148950099945e-07
```
[the remark this answers is not in the document]
Comment (StanczakDominik) [reply plasmapy/formulary/mathematics.py:142]: This results from the fact that I messed with the exact formulation of the function a little, to a form which is supposedly more stable numerically.

File: plasmapy/formulary/tests/test_Chandrasekhar_G.py [hunk 2]
```diff
@@ -1,8 +1,32 @@
...
+    x=st.floats(
+        allow_nan=False,
+        allow_infinity=False,
+    )
+)
+@example(x=np.finfo(np.float64).eps)
+@example(x=np.finfo(np.float64).max)
+@example(x=np.finfo(np.float64).min)
+@example(x=0)
+@example(x=3.761264e-20)
```
[the remark this answers is not in the document]
Comment (StanczakDominik) [reply plasmapy/formulary/tests/test_Chandrasekhar_G.py:20]: I simply added them as I knew they'd be potentially problematic. No harm in keeping them, I think :)

**Comment by StanczakDominik** [comment]
pre-commit.ci autofix

**Comment by StanczakDominik** [comment]
Thanks for the review @namurphy !

**Pull request merged**